staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -81,30 +81,30 @@ func newPatcher(o *ApplyOptions, info *resource.Info) (*Patcher, error) {
 	}
 
 	helper := resource.NewHelper(info.Client, info.Mapping)
-	if o.ServerDryRun {
+	if o.DryRunStrategy == cmdutil.DryRunServer {
 		if err := resource.VerifyDryRun(info.Mapping.GroupVersionKind, o.DynamicClient, o.DiscoveryClient); err != nil {
 			return nil, err
 		}
-		helper.DryRun(o.ServerDryRun)
+		helper.DryRun(true)
 	}
 	return &Patcher{
-		Mapping:       info.Mapping,
-		Helper:        helper,
-		DynamicClient: o.DynamicClient,
-		Overwrite:     o.Overwrite,
-		BackOff:       clockwork.NewRealClock(),
-		Force:         o.DeleteOptions.ForceDeletion,
-		Cascade:       o.DeleteOptions.Cascade,
-		Timeout:       o.DeleteOptions.Timeout,
-		GracePeriod:   o.DeleteOptions.GracePeriod,
-		ServerDryRun:  o.ServerDryRun,
-		OpenapiSchema: openapiSchema,
-		Retries:       maxPatchRetry,
+		Mapping:        info.Mapping,
+		Helper:         helper,
+		DynamicClient:  o.DynamicClient,
+		Overwrite:      o.Overwrite,
+		BackOff:        clockwork.NewRealClock(),
+		Force:          o.DeleteOptions.ForceDeletion,
+		Cascade:        o.DeleteOptions.Cascade,
+		Timeout:        o.DeleteOptions.Timeout,
+		GracePeriod:    o.DeleteOptions.GracePeriod,
+		DryRunStrategy: o.DryRunStrategy,
```

Comment:
I don't know about this one? The patcher doesn't really need to know if it's client or server so I think this only makes things more complicated.

Reply:
Oh right, reverted